src/platform/stm32f1/spi.c

```diff
@@ -3,22 +3,18 @@
  * SPDX-FileCopyrightText: 2021 Rafael Silva <silvagracarafael@gmail.com>
  */
 
-#include "platform/stm32f1/spi.h"
+#include <stm32f1xx.h>
+
 #include "platform/stm32f1/gpio.h"
 #include "platform/stm32f1/rcc.h"
+#include "platform/stm32f1/spi.h"
 #include "util/types.h"
 
-int spi_init(
-	struct spi_drv_data_t *drv_data,
-	enum spi_interface_no interface_no,
-	enum spi_mode mode,
-	u32 frequency,
-	u8 bit_order,
-	struct gpio_pin_t cs_gpio,
-	u8 cs_inverted)
+void *spi_init_interface(enum spi_interface_no interface_no, enum spi_mode mode, u32 frequency, u8 bit_order)
```

Comment:
Maybe we can call this `spi_device_init` and leave the other `spi_init`. This function would be used to create the structure and the other one to actually init the SPI interface/device.

Please make it return a `spi_device_t`.

Reply:
Nevermind, I read your comment and it make sense.

Could we have a `spi_interface_t` and `spi_device_t`, make the device get an interface and then `spi_interface_init` and `spi_device_init`?

```c
struct spi_interface_t {...};
struct spi_device_t {
	struct spi_interface_t interface;
	...
};
```

```c
struct spi_interface_t interface = spi_interface_init(...);
struct spi_device_t spi_sensor = spi_device_init(interface, ...);
struct spi_device_t spi_memory = spi_device_init(interface, ...);
```

Reply:
Not sure I completely agree with this, but I see somethings that could change.
How about we have this `spi_init_interface` as is, except it just returns success or error, and a `spi_device_init` that also receives a `interface_no` instead of the pointer to the interface, and returns a `spi_device_t`

Reply:
oh, didnt see your second comment, didnt update the page, my response was regarding the first comment only

Reply:
didnt quite do what you suggested, but I think it's good how it is right now